### PR TITLE
anthropic[patch]: use tool calls if content is empty

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/partners/anthropic/langchain_anthropic/chat_models.py [hunk 1]
```diff
@@ -677,6 +688,29 @@ def _tools_in_params(params: dict) -> bool:
...
+    name: str
+    input: dict
+    id: str
+
+
+def _lc_tool_calls_to_anthropic_tool_use_blocks(
+    tool_calls: List[ToolCall],
+) -> List[_AnthropicToolUse]:
+    blocks = []
+    for tool_call in tool_calls:
```
Comment (ccurme) [inline libs/partners/anthropic/langchain_anthropic/chat_models.py:702]: Nuno had a comment on this: https://github.com/langchain-ai/langchain/pull/20519/files#r1567919816
> can we support putting string content, if not empty, as an additional part of type text?

**Review by ccurme — APPROVED** [review]

File: libs/partners/anthropic/langchain_anthropic/chat_models.py [hunk 2]
```diff
@@ -195,6 +197,15 @@ def _format_messages(messages: List[BaseMessage]) -> Tuple[Optional[str], List[D
                     raise ValueError(
                         f"Content items must be str or dict, instead was: {type(item)}"
                     )
+        elif (
+            isinstance(message, AIMessage)
+            and not message.content
+            and message.tool_calls
```
[the remark this answers is not in the document]
Comment (baskaryan) [reply libs/partners/anthropic/langchain_anthropic/chat_models.py:203]: watcha mean

File: libs/partners/anthropic/tests/unit_tests/test_chat_models.py [hunk 3]
```diff
@@ -268,3 +272,75 @@ def test_convert_to_anthropic_tool(
...
+
+def test__format_messages_with_content_and_tool_calls() -> None:
+    system = SystemMessage("fuzz")
+    human = HumanMessage("foo")
+    # If content and tool_calls are specified, content is preferred.
+    ai = AIMessage(
+        "not a tool call",
+        tool_calls=[{"name": "bar", "id": "1", "args": {"baz": "buzz"}}],
+    )
+    tool = ToolMessage(
```
Comment (efriis) [inline libs/partners/anthropic/tests/unit_tests/test_chat_models.py:324]:
```\
    ai = AIMessage(
        [{text content}, {tool call content matching below}, {text content}],
        tool_calls=[{"name": "bar", "id": "1", "args": {"baz": "buzz"}}],
    )
    tool = ToolMessage(
```
should we add a case like this? should theoretically dedupe right?

Comment (baskaryan) [reply libs/partners/anthropic/tests/unit_tests/test_chat_models.py]: currently ignores tool_calls if content is a list

**Pull request merged**